test/run-tests

```diff
@@ -439,98 +440,68 @@ class Task:
-    comments = get_comments(gh, owner, repo, pull)
+    comments = pyghpull.get_issue_comments()
     whitelist_command = f"[citest commit:{head}]"
     for comment in comments:
-        if comment["author_association"] == "MEMBER":
-            if whitelist_command in comment["body"]:
+        if comment.raw_data["author_association"] == "MEMBER":
+            if whitelist_command in comment.body:
                 logging.debug("PR is ok to test - whitelisted by member")
                 return True
 
-    result = gh.get(f"repos/{owner}/{repo}/issues/{pull}/labels")
-    if result.status_code == 200 and "needs-ci" in (x["name"] for x in result.json()):
-        logging.debug("PR is ok to test - has needs-ci label")
-        return True
+    labels = pyghpull.get_labels()
+    for label in labels:
+        if "needs-ci" in label.name:
+            logging.debug("PR is ok to test - has needs-ci label")
+            return True
+
     logging.debug("PR is not ok to test")
+    return False
 
 
-def get_statuses(gh, owner, repo, sha, max_num_statuses):
+def get_statuses(commits):
     """
     Fetches all statuses of the given commit in a repository and returns a dict
     mapping context to its most recent status.
     Will return at most max_num_statuses.   By default, github will return 30, but
     some of our PRs have more than that.
     """
-    # https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref
-    status = gh.get(
-        f"repos/{owner}/{repo}/commits/{sha}/status?per_page={max_num_statuses}"
-    ).json()
-    if status["total_count"] > len(status["statuses"]):
-        # error - there are too many statuses
-        if max_num_statuses <= len(status["statuses"]):
-            logging.error(
-                f"Total number of statuses {status['total_count']} exceeds "
-                f"max_num_statuses {max_num_statuses} statuses for "
-                f"{owner}/{repo}/commits/{sha} - please increase max_num_statuses"
-            )
-        else:
-            # hit hard limit - will have to use paging :-(
-            logging.error(
-                f"Total number of statuses {status['total_count']} exceeds "
-                f"max_num_statuses {max_num_statuses} statuses for "
-                f"{owner}/{repo}/commits/{sha} - need to use paging"
-            )
-    return {x["context"]: x for x in status["statuses"]}
-
-
-def get_comments(gh, owner, repo, pullnr, after=""):
-    """ Get comments for a pull request, optionally after a certain time """
-    result = gh.get(f"repos/{owner}/{repo}/issues/{pullnr}/comments")
-    comments = []
-    if result.status_code == 200:
-        comments = result.json()
-
-    new_comments = []
-    for comment in comments:
-        comment_update = comment["updated_at"]
-        if comment_update > after:
-            new_comments.append(comment)
-    comments = new_comments
+    statues = []
+    for commit in commits:
```

Comment:
This is for multiple commits?  Not just `head`?

Reply:
Ah, you are right... Somehow, I was thinking I should look into all, but I should not have. Let me fix it.